### ISSUE TITLE
More helpful ConnectionError message on query timeout

### ISSUE THREAD

**Issue by novaugust** [elixir-ecto/db_connection#119]
### Description of issue

When a connection errors due to a query timeout a helpful message is logged to errors like
>  Postgrex.Protocol (#PID<0.603.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.14026.0> timed out because it checked out the connection for longer than 15000ms

Whatever process was using that connection will also exit if it wasn't trapping, which the error_logger will pick up and print but with a far less helpful message
```
#PID<0.14026.0> running ... terminated 
(exit) an exception was raised:     ** (DBConnection.ConnectionError) ssl recv: closed        
```
When reporting exceptions to a service, only this second, less useful message gets reported: "ssl recv: closed". Devs then have to open up the logs and go searching for a Postgrex error logged sometime before the exception that contains a matching pid.

### Request

Would it be possible to have these timeout exits maintain the useful "timed out because..." message as their exception message? It'd be a big quality of life improvement as a developer, both to cut down on the exception dereferencing but _also_ to differentiate timeout errors from _actual_ ssl errors (handshake, network, ...)

**Comment by fishcakez** [comment]
This isn't possible in the current implementation. We can likely improve it after #108 if we move to using ETS where we currently use pdict.

**Comment by josevalim** [comment]
To test this, we need to sleep on a execute instruction and make sure we provide a good message.

**Comment by josevalim** [comment]
We should also probably include a bit of the client information in the failure.

**Comment by josevalim** [comment]
@fishcakez it seems the major concern from @novaugust is not the execute timeout message, which we already handle it nicely (although we could include client information in there) but rather that the process that triggers the timeout will fail with either tcp/ssl recv error. To be fair, I don't think we can fix this though.

**Comment by josevalim** [comment]
Closed in favor of #157.

**Comment by novaugust** [comment]
in the words of jose:

❤️ 💚 💙 💛 💜
